### PR TITLE
Fips documentation

### PR DESCRIPTION
**Pull request by briandowns**

### CODE REVIEW

File: docs/fips_support.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Rancher Kubernetes Enginer 2
+
+## Support for FIPS Cryptography
+
+Starting with Rancher Kubernetes Enginer version 2 (RKE2), you can install a cluster that uses FIPS validated libraries.
```
Comment (galal-hussein) [inline docs/fips_support.md:5]: nit: typo in Enginer --> Engine

File: docs/fips_support.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Most of the components of the RKE2 system are statically compiled with the GoBoring Go compiler implementation that takes advantage of the BoringSSL library. RKE2, from a component perspective, is broken up in a number of sections. The list below contains the sections and associated components.
+
+* etcd
+
+* Kubernetes
+  * API Server
+  * Controller Manager
+  * Scheduler
+  * Kubelet
+  * KubeProxy
```
Comment (galal-hussein) [inline docs/fips_support.md:23]: We should mention i tihnk that this particular component (kubeproxy) is separated in its own image

File: docs/fips_support.md (hunk 1 above)
Comment (briandowns) [reply docs/fips_support.md:5]: Resolved. Will push all changes at once. Thanks!

File: docs/fips_support.md (hunk 2 above)
Comment (briandowns) [reply docs/fips_support.md:23]: Does that matter in this context?

Comment (galal-hussein) [reply docs/fips_support.md:23]: I guess not, resolving this one, thanks!

File: docs/fips_support.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Rancher Kubernetes Enginer 2
```
Comment (davidnuzik) [inline docs/fips_support.md:1]: SP: "Enginer"

File: docs/fips_support.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  * Kubectl
+
+* Helm
+  * Flannel
+  * Calico
+  * CoreDNS
+
+## Runtime
+
+To ensure that all aspects of the system architecture are using FIPS 140-2 compliant algorithm implemenations, the RKE2 runtime contains utilities statically compiled with the customized Go compiler for FIPS 140-2 compliance. This ensures that all levels of the stack are compliant from Kuberenetes daemons to container orchestration mechanics.
```
Comment (davidnuzik) [inline docs/fips_support.md:34]: SP: "implemenations" and "Kuberenetes"

File: docs/fips_support.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  * containerd-shim
+  * containerd-shim-runc-v1
+  * containerd-shim-runc-v2
+* ctr
+* runc
+* crictl
+
+## Ingress
+
+Ingress is not included in the RKE2 FIPS 140-2 compliance purview. This is the responsibility of the customers as ingress is ultimately their choice of implemenation.
```
Comment (davidnuzik) [inline docs/fips_support.md:46]: SP: "implemenation"

Comment (davidnuzik) [inline docs/fips_support.md:46]: Nit: Sorry, this is quite picky. Suggest to change "customers" to "user" -- just thinking captures wider audience even though not super likely many non-customers will be caring about FIPS.

**Comment by briandowns** [comment]
Now looking for a VSCode spell checker... :/

**Pull request merged**